xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.h

```diff
@@ -83,6 +84,8 @@ class CVideoBufferCopy : public CVideoBufferShared
       : CVideoBufferShared(id) {}
 
   Microsoft::WRL::ComPtr<ID3D11Resource> m_copyRes;
+  Microsoft::WRL::ComPtr<ID3D11Resource> m_pResource;
+  Microsoft::WRL::ComPtr<ID3D11DeviceContext> m_pDeviceContext;
```

Comment:
no need to keep ref to d3d11 device. it always can be gotten from d3d11 resource.
the same for `m_pResource` it's used once in `CVideoBufferShared::Initialize`

Reply:
I do that for performance reasons: Is not need obtain these ComPtr for every frame decoded (24 or 60 times every second). Can be obtained one time at beginning and store it.

NOTE: `CVideoBufferShared::Initialize`  despite his name is called for every frame in loop.

Reply:
there is no performance penalty, this is not a creation of new device instance, just getting a reference which (I think already stored internally) and increasing reference count.

Reply:
You are probably right and the actual impact is minimal (for a current PC there is no effort to do something simple 60 times per second) but I would still like to remove this block of code from the loop (more for code cleanliness than performance):

```
  ComPtr<ID3D11Resource> pResource;
  ComPtr<ID3D11Device> pDevice;
  ComPtr<ID3D11DeviceContext> pDeviceContext;

  if (FAILED(CVideoBuffer::GetResource(&pResource)))
  {
    CLog::LogF(LOGDEBUG, "unable to get decoder resource");
    return;
  }

  // decoder ctx
  decoder->m_pD3D11Context->GetDevice(&pDevice);
  pDevice->GetImmediateContext(&pDeviceContext);
```

For my taste I like loops as simple as possible and without logging messages, leaving only the minimum essential that here is the texture copy itself:
```
    // sends commands to GPU (ensures that the last decoded image is ready)
    m_pDeviceContext->Flush();

    m_pDeviceContext->CopySubresourceRegion(m_copyRes.Get(), 0, 0, 0, 0, m_pResource.Get(),
                                            CVideoBuffer::GetIdx(), nullptr);
```

For me it makes sense to store at the beginning `m_pResource` and `m_pDeviceContext` because are used inside the loop.

Still if you think it is not acceptable because it has some drawbacks that I have not taken into account I would like to know the exact reason.